version.go

```diff
@@ -405,3 +406,25 @@ func (v *Version) UnmarshalText(b []byte) error {
 func (v *Version) MarshalText() ([]byte, error) {
 	return []byte(v.String()), nil
 }
+
+// Scan implements the sql.Scanner interface.
+func (v *Version) Scan(src interface{}) error {
+	switch src := src.(type) {
+	case *Version:
+		*v = *src
+		return nil
+	case []byte:
+		return v.UnmarshalText(src)
+	case string:
+		return v.UnmarshalText([]byte(src))
```

Comment:
Could you please provide any rationale behind the choice of types to support here? Is this based on how different DB engines and Go drivers encode the data? If so - can you link to any of those?

I'm not particularly opposed to any of them but want to understand the scope and intentions here. My - perhaps wrong - assumption is that as long as the `Value()` method below is used, then we may only expect `string` here and not bytes or native `*Version` type?

Reply:
I think I got a little bit carried away here 😏 Removed all but the `string` and `nil` cases 👍🏻